git_theta/params.py

```diff
@@ -1,46 +1,106 @@
-"""Utilties to summarize parameters."""
-
-import hashlib
-import numpy as np
-
-
-def get_shape_str(p):
-    """
-    Parameters
-    ----------
-    p : list or scalar
-
-    Returns
-    -------
-    str
-        shape of parameter
-    """
-    return str(np.asarray(p).shape)
-
-
-def get_dtype_str(p):
-    """
-    Parameters
-    ----------
-    p : list or scalar
-
-    Returns
-    -------
-    str
-        dtype of parameter
-    """
-    return np.asarray(p).dtype.str
-
-
-def get_hash(p):
-    """
-    Parameters
-    ----------
-    p : list or scalar
-
-    Returns
-    -------
-    str
-        hash of parameter bytes
-    """
-    return hashlib.sha1(np.asarray(p).tobytes()).hexdigest()
+"""Classes for serializing model updates."""
+
+from collections import defaultdict
+import tensorstore as ts
+import io
+import tarfile
+import posixpath
```

Comment:
I believe tar can handle windows style paths as well. Posixpath is explicitly used here since I wanted to make sure the virtual paths in the tar file are consistent when people read and write them from both unix and windows. If the code used os.path instead then windows users wouldn't be able to properly split the virtual paths created by unix users and vice-versa.